### PR TITLE
autorestic 1.0.8 (new formula)

### PR DESCRIPTION
**Pull request by cupcakearmy**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

It's my first fomula, but i think it should be fine.

### CODE REVIEW

**Comment by cupcakearmy** [comment]
Associated issue https://github.com/cupcakearmy/autorestic/issues/58

File: Formula/autorestic.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  head "https://github.com/cupcakearmy/autorestic.git"
+
+
+  depends_on "go" => :build
+  depends_on "restic"
+
+  def install
+    ENV["GOPATH"] = HOMEBREW_CACHE/"go_cache"
+
+    system "go", "build", "-o", "autorestic", "./main.go"
```
Comment (SMillerDev) [inline Formula/autorestic.rb:16]:
```suggestion
    system "go", "build", *std_go_args "./main.go"
```

File: Formula/autorestic.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    
+    mkdir "completions"
+    system "bash", "-c", "./autorestic completion bash > completions/autorestic_bash"
+    system "bash", "-c", "./autorestic completion zsh > completions/autorestic_zsh"
+    system "bash", "-c", "./autorestic completion fish > completions/autorestic_fish"
+    bash_completion.install "completions/autorestic_bash"
+    zsh_completion.install "completions/autorestic_zsh"
+    fish_completion.install "completions/autorestic_fish"
+
+    bin.install "autorestic"
```
[the remark this answers is not in the document]
Comment (carlocab) [reply Formula/autorestic.rb:26]: Doing `bin.install` shouldn't be needed once you use `std_go_args`.

**Comment by cupcakearmy** [comment]
Thanks for all the quick feedback. I'll update the pr in few hours :)

File: Formula/autorestic.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  license "Apache-2.0"
+  head "https://github.com/cupcakearmy/autorestic.git"
+
+
+  depends_on "go" => :build
+  depends_on "restic"
+
+  def install
+    ENV["GOPATH"] = HOMEBREW_CACHE/"go_cache"
+
```
[the remark this answers is not in the document]
Comment (cupcakearmy) [reply Formula/autorestic.rb:15]: You are right :) just tested it

File: Formula/autorestic.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    system "go", "build", *std_go_args "./main.go"
+    
+    bin.install "autorestic"
```
[the remark this answers is not in the document]
Comment (cupcakearmy) [reply Formula/autorestic.rb:18]: Yes, was writing the test :)

**Comment by cupcakearmy** [comment]
@carlocab i've addressed all the things (i think)
Thanks for all the input ❤️

File: Formula/autorestic.rb [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    }
+    File.open(testpath/".autorestic.yml", "w") { |file| file.write(config.to_yaml) }
+    # Test data
+    (testpath/"repo"/"test.txt").write("This is a testfile")
+    # Backup
+    system "#{bin}/autorestic", "backup", "-a"
+    # Restore
+    system "#{bin}/autorestic", "restore", "-l", "foo", "--to", "restore"
+    # Assert same
+    assert compare_file testpath/"repo"/"test.txt", testpath/"restore"/"test.txt"
```
Comment (nandahkrishna) [inline Formula/autorestic.rb:36]:
```suggestion
    (testpath/"repo"/"test.txt").write("This is a testfile")
    system "#{bin}/autorestic", "backup", "-a"
    system "#{bin}/autorestic", "restore", "-l", "foo", "--to", "restore"
    assert compare_file testpath/"repo"/"test.txt", testpath/"restore"/"test.txt"
```

The comments aren't necessary here.

**Review by nandahkrishna — COMMENTED** [review]
Thanks @cupcakearmy, some minor suggestions:
* We can remove the extra comments from the `test` block.
* Please use `brew style --fix autorestic` to correct all the style errors here automatically. If there are any errors that aren't autocorrected, please ensure you fix them too.

Once you've made these changes, please squash all your commits into a single commit with the message:
```
autorestic 1.0.8 (new formula)
```

File: Formula/autorestic.rb [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    system "go", "build", *std_go_args, "./main.go"
+    
+    (bash_completion/"autorestic").write Utils.safe_popen_read("#{bin}/autorestic", "completion", "bash")
+    (zsh_completion/"_autorestic").write Utils.safe_popen_read("#{bin}/autorestic", "completion", "zsh")
+    (fish_completion/"autorestic.fish").write Utils.safe_popen_read("#{bin}/autorestic", "completion", "fish")
+  end
+
+  test do
+    require 'yaml'
+    # Create sample config
```
Comment (nandahkrishna) [inline Formula/autorestic.rb:23]:
```suggestion
```

**Comment by cupcakearmy** [comment]
@nandahkrishna How do I squash commits? I tried a few things but commits keep getting more and not less 😅🤦‍♂️

**Comment by carlocab** [comment]
You want to do `git rebase`. As a rule, when working on homebrew/core PRs, always do `git rebase` instead of `git merge`. `git merge` creates merge commits, which @BrewTestBot does not like (hence the `automerge-skip` label earlier).

I fixed this for you this time around.

**Comment by cupcakearmy** [comment]
> You want to do `git rebase`. As a rule, when working on homebrew/core PRs, always do `git rebase` instead of `git merge`. `git merge` creates merge commits, which @BrewTestBot does not like (hence the `automerge-skip` label earlier).
> 
> I fixed this for you this time around.

Thanks you so much :) I was rebasing but probabl not the correct way around xD

**Comment by nandahkrishna** [comment]
There seem to have been a few problems again, so I squashed all your commits and pushed them.

**Review by nandahkrishna — DISMISSED** [review]
Thanks @cupcakearmy!

**Comment by cupcakearmy** [comment]
@nandahkrishna I just opened vscode and it synced the repo 🤦‍♂️ now it's 19 commits again, I'm really sry.

**Comment by nandahkrishna** [comment]
I've fixed it 👍🏼. In the future, please create pull requests from a new branch rather than the default branch (`master` in this case).

**Comment by carlocab** [comment]
Please read the Homebrew's [guide](https://docs.brew.sh/How-To-Open-a-Homebrew-Pull-Request) on opening pull requests. This is also referenced in the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md) that you read before opening your pull request.

**Review by nandahkrishna — APPROVED** [review]

**Comment by cupcakearmy** [comment]
@SMillerDev i think your review is still required :)

**Pull request closed without merging**